src/main/java/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -276,6 +295,23 @@ open class ParallelThreadsRunner(
     override fun createTransformer(cv: ClassVisitor): ClassVisitor {
         return CancellabilitySupportClassTransformer(cv)
     }
+
+    override fun constructStateRepresentation(): String? =
+        stateRepresentationMethod?.let{ getMethod(testInstance, it) }?.invoke(testInstance) as String?
+
+    override fun transformTestClass() {
+        super.transformTestClass()
+        initializeTestThreadExecutions()
+    }
+
+    private fun initializeTestThreadExecutions() {
```

Comment:
Please, move the initialization back to the constructor, you do not need to re-create them.

Reply:
However, it could be better to keep a method `createTestThreadExecutions` so that the code becomes cleaner